lib/client.js

```diff
@@ -2060,6 +2066,145 @@ MatrixClient.prototype.paginateEventContext = function(eventContext, opts) {
     return promise;
 };
 
+/**
+ * Get an EventTimeline for the given event
+ *
+ * <p>If the room object already has the given event in its store, the
+ * corresponding timeline will be returned. Otherwise, we make a /context
+ * request, and use it to construct an EventTimeline.
+ *
+ * @param {Room} room  The room to look for the event in
+ * @param {string} eventId  The ID of the event to look for
+ *
+ * @return {module:client.Promise} Resolves:
+ *    {@link module:models/event-timeline~EventTimeline} including the given
+ *    event
+ */
+MatrixClient.prototype.getEventTimeline = function(room, eventId) {
+    // don't allow any timeline support unless it's been enabled.
+    if (!this.timelineSupport) {
+        throw Error("timeline support is disabled. Set the 'timelineSupport'" +
+                    " parameter to true when creating MatrixClient to enable" +
+                    " it.");
+    }
+
+    if (room.getTimelineForEvent(eventId)) {
+        return q(room.getTimelineForEvent(eventId));
+    }
+
+    var path = utils.encodeUri(
+        "/rooms/$roomId/context/$eventId", {
+            $roomId: room.roomId,
+            $eventId: eventId,
+        }
+    );
+
+    // TODO: we should implement a backoff (as per scrollback()) to deal more
+    // nicely with HTTP errors.
+    var self = this;
+    var promise =
+        self._http.authedRequest(undefined, "GET", path
+    ).then(function(res) {
+        if (!res.event) {
+            throw Error("'event' not in '/context' result - homeserver too old?");
```

Comment:
Please always `new` your constructors. This will still work for `Error` objects due to defined factory functions but we're generally consistent using `new` everywhere.